app/repositories/plip_repository.rb

```diff
@@ -7,6 +7,15 @@ def current_plip
     plips.last
   end
 
+  def find_plip_by_slug(slug)
+    phase = Cycle.find_by_slug(slug).phases.last
```

Comment:
Though cycles which contain a plip are always created with a single phase and the petition one turns out to be the last one, we can not assume that. What you should do here instead is find the first ohase which is linked with a petition plugin.